### PR TITLE
Fixes for following formats: vtk, medit, med

### PR DESCRIPTION
**Pull request by vlukes**
### VTK

- saving call data was not correctly implemented, tags `CELL_DATA` and `FIELD FieldData` should be in VTK file only once
- limit to `FeildData` dimension has no reasons

### MEDIT

- `"medit:ref"` for cells must be a list of arrays containing group ids

### MED

- `FAS` object  can be located in the mesh object, not only in the root

### CODE REVIEW

File: meshio/vtk/_vtk.py [hunk 1]
```diff
@@ -629,10 +629,10 @@ def pad(array):
             _write_field_data(f, mesh.point_data, binary)
 
         # write cell data
-        for name, data in mesh.cell_data.items():
-            total_num_cells = sum([len(c) for c in data])
+        if mesh.cell_data:
+            total_num_cells = sum(len(c.data) for c in mesh.cells)
             f.write(f"CELL_DATA {total_num_cells}\n".encode("utf-8"))
-            _write_field_data(f, raw_from_cell_data(mesh.cell_data), binary)
+            _write_field_data(f, mesh.cell_data, binary)
```
Comment (nschloe) [inline meshio/vtk/_vtk.py]: Can we really just write `mesh.cell_data`? I should be a dict of lists of numpy.arrays, really. `raw_from` converts this to a dict of numpy.array by concatenation.

Comment (vlukes) [reply meshio/vtk/_vtk.py]: Yes, `mesh.cell_data` is a dict of lists of arrays and the lists must be joined. This is done in  `_write_field_data()`, if `data` is a list. This produces just one `CELL_DATA` + `FIELD` section followed by the fields.

There was probably something strange in the commit, I didn't see the changes in `_write_field_data()`, after re-pushing, I see it again.

**Comment by vlukes** [comment]
The fix for vtk should solve #643.

**Comment by gdmcbain** [comment]
Oh, I thought the bug #643 was in reading MSH 2.2 rather than the subsequent write to VTK.

**Comment by vlukes** [comment]
Yes, I see, there is another issue. In the MSH input file in #643, the points are not numbered in a continuous range.

File: meshio/vtk/_vtk.py (hunk 1 above)
Comment (nschloe) [reply meshio/vtk/_vtk.py]: Alright, since the tests pass, let's give it a go.

**Pull request merged**